File: Data Structure/C++/2.Stack/stack implementation using array.cpp
#include<iostream>
#include<cstdlib> // for clear scr



using namespace std;

#define m 10
class stack
{
    int top=-1;

    public:
        int arr[m];

        void push();
        bool isFull();
        void pop();
        bool isEmpty();
        void display();
};

bool stack :: isEmpty()
{
    if(top == -1)
        return true;
    else
        return false;
}

bool stack :: isFull()
{
    if(top == m -1)
        return true;
    else
        return false;
}

void stack :: push()
{
    int item;
    if(isFull())
    {
        cout<<"Stack is Overflow\n";
        return;
    }
    cout<<"Enter an item\n";
    cin>>item;

    arr[++top] = item;
    cout<<item <<" is inserted"<<endl;
}

void stack :: pop()
{
    if(isEmpty())
    {
        cout<<"Stack is UnderFlow\n";
        return;
    }
    cout<<arr[top--]<<" is popped"<<endl;
}

void stack :: display()
{

    if(isEmpty())
    {
        cout<<"Stack is UnderFlow\n";
        return;
    }

    for(int i=0;i<=top;i++)
    {
        cout<<arr[i]<<"  ";
    }
    cout<<endl;

}


int main()
{
    stack obj;
    int ch;

    do
    {
        cout<<"1.Push\n2.Pop\n3.Traverse\n4.Exit\nEnter your choice\n";
        cin>>ch;

        system("cls");

        switch(ch)
        {
            case 1:
                obj.push();
                break;

            case 2:
                obj.pop();
                break;

            case 3:
                obj.display();
                break;

            case 4:
                cout<<"Exiting..."<<endl;
                break;

            default:
                cout<<"Enter the correct input"<<endl;
        }
        }while(ch!=4);

}

File: CodePractices/src/Stack/StackImplementationUsingArray.java
package Stack;

import java.util.*;
public class StackImplementationUsingArray {
    static int top = -1;
    static int cap = 5;
    static int[] st = new int[cap];

    private static boolean push(int item){
        if(top==cap-1){
            return false;
        }
        else{
            st[++top] = item;
            return true;
        }
    }

    private static boolean pop(){
        if(top==-1){
            return false;
        }
        else{
            top--;
            return true;
        }
    }

    private static void print(){
        for(int i=0;i<=top;i++){
            System.out.print(st[i]+" ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        while(true){
            System.out.println("1.Push");
            System.out.println("2.Pop");
            System.out.println("3.Print");
            System.out.println("4.Exit");

            int ch = new Scanner(System.in).nextInt();

            switch (ch){
                case 1:
                    System.out.println("Enter item to push:");
                    int item = new Scanner(System.in).nextInt();
                    if(push(item))
                        System.out.println(item+ " pushed!!!");
                    else
                        System.out.println("Stack overflow!!!");
                    break;
                case 2:
                    if(pop())
                        System.out.println(st[top+1]+" popped!!!");
                    else
                        System.out.println("stack underflow!!!");
                    break;
                case 3:
                    if(top==-1){
                        System.out.println("Stack underflow!!!");
                    }else {
                        System.out.println("Stack elements are: ");
                        print();
                    }
                    break;
                case 4:
                    System.exit(1);
                default:
                    System.out.println("You have not entered valid input data!!! Try again");

            }

        }
    }
}
